### ISSUE TITLE
signalingClientConnectSync fails in arm 32 bit environment

### ISSUE THREAD

**Issue by ronievron** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#954]
Code works fine in standard Linux 64 bit environment.  But, when built for 32 bit arm, the code fails.
Code and log files attached.
[failed.log](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/5547956/failed.log)
[success.log](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/5547957/success.log)
[code.zip](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/5548009/code.zip)

**Comment by lherman-cs** [comment]
@ronievron I'm not sure if the issue is related to 32 bits vs 64 bits. From your logs, it shows that [libwebsockets](https://github.com/warmcat/libwebsockets) fired an error after the connection establishment. But, this error event was coming from libwebsockets. So, even if there's an issue with 32 bits arm, there's nothing we can do from our side. Could you check if your 32 bit machine has a good network condition?

**Comment by MushMal** [comment]
@ronievron you provided very little information to help us help you to debug your issue.

* Looking at the logs, I can see that the SDK is able to create the client (calling multiple backend APIs) then it's failing and retrying to connect and eventually times out. There is no information why it fails so you might want to compile the libWebSockets in debug mode to get more spew at the network level and enable verbose logs in the SDK side. Search the issues for more information on how to enable it.
* Your sample application looks OK but we don't recommend casting the handle to internally defined objects. This can change with every minor commit of the code. Use the public APIs to access the ice server configuration. There are two - one that returns the count of objects and the other to retrieve the ice server config objects themselves. Take a look at the sample to understand how it's used.
* Include the platform information so we can try to understand what 32/64 bit ARM platform you are using and what OS/version.
* In your sample, try to set `channelInfo.asyncIceServerConfig = FALSE;` as not all platforms support dual context execution in parallel.
* Use stock master sample and see how that works for you. We won't be able to help much with your own code.

**Comment by disa6302** [comment]
Closing since this has gone stale. Feel free to reopen or create a new issue if you have anymore details
